src/deserialize/parse.rs

```diff
@@ -30,11 +30,19 @@ pub(crate) fn parse_edn(c: Option<char>, chars: &mut std::str::Chars) -> Result<
     }
 }
 
-fn read_key(chars: &mut std::str::Chars) -> Edn {
-    let c_len = chars
+fn read_key_or_nsmap(chars: &mut std::str::Chars) -> Result<Edn, Error> {
+    let mut key_chars = chars
         .clone()
```

Comment:
Why is this clone here again? I mean, there's a new one down there :thinking:

Reply:
because `take_while` takes the value it matched wrongly

Reply:
All take_whiles are inside clones

Reply:
Yeah, but there's a new one on the function now, can we remove one of those, or is it necessary?

Reply:
it is necessary, and we removed from here 
```rust
fn read_key(chars: &mut std::str::Chars, c_len: usize) -> Edn {
    let mut key = String::from(":");
    let key_chars = chars.take(c_len).collect::<String>();
    key.push_str(&key_chars);
    Edn::Key(key)
}
```